src/capi/object.cpp

```diff
@@ -614,6 +614,40 @@ extern "C" PyObject** _PyObject_GetDictPtr(PyObject* obj) noexcept {
     }
 }
 
+// This function returns the copy of object's dict, not a writable dict.
+extern "C" PyObject* PyObject_GetDict(PyObject* obj) noexcept {
+    PyTypeObject* tp = Py_TYPE(obj);
+    if (!tp->instancesHaveHCAttrs()) {
+        if (!tp->instancesHaveDictAttrs())
+            return NULL;
+        return obj->getDict();
```

Comment:
I think this will not always return a copy of the dict  (the `obj->getDict();` case)
How do you feel about naming it `PyObject_GetDictCopy`? Or does some other python implementation already use `PyObject_GetDict`?